### DISCUSSION TITLE
Good way to visually break up a list/set/dict of strings each split across multiple lines?

### DISCUSSION THREAD

**Discussion by edbennett** [astral-sh/ruff#12147] [Q&A]
[PLR2044](https://docs.astral.sh/ruff/rules/empty-comment/) having been finalised has raised an issue in a project I work on.

I have some dicts of long strings, each of which is broken across multiple lines.

```
_COLUMN_DESCRIPTIONS = {
    #
    "MC_Time": "Index enumerating the Monte Carlo samples.",
    #
    "Time": "Physical euclidean time coordinate "
    "along which correlation is measured.",
    #
    "Internal": "Any further internal structure, e.g.,"
    "an index enumerating interpolating operators, "
    "a blocking or smearing level, "
    "or any combination thereof.",
    #
    "Correlation": "Measured values of the correlators.",
    #
    "Vac_exp": "Measured values of the vacuum expectation values (VEVs).",
}
```

Empty comments are used to separate entries, as otherwise it's not obvious where strings end—the trailing `,` is not visually striking when every line ends with a `"` and many have other punctuation before that. There isn't really a need for a non-empty comment; each dict entry is self-explanatory.

Is there a recommended alternative for this? If not, then I can add PLR2044 to my exceptions list and/or `noqa` these data structures, but if there are better ways to format this then I'd be enthusiastic to adopt them.

**Comment by MichaReiser** [comment]
I think that depends on if you use the formatter or not. The way I would format it and I would like `ruff format ` to format it in the future is:

```python
_COLUMN_DESCRIPTIONS = {
    "MC_Time": "Index enumerating the Monte Carlo samples.", 
    "Time": "Physical euclidean time coordinate "
        "along which correlation is measured.",
    "Internal": "Any further internal structure, e.g.,"
        "an index enumerating interpolating operators, "
        "a blocking or smearing level, "
        "or any combination thereof.",
    "Correlation": "Measured values of the correlators.",
    "Vac_exp": "Measured values of the vacuum expectation values (VEVs).",
}
```

Possibly with some empty lines in between. 

An alternative that is also `ruff format` compatible is to use parentheses, which is what Black 24+ does:

```python
_COLUMN_DESCRIPTIONS = {
    "MC_Time": "Index enumerating the Monte Carlo samples.",
    "Time": (
        "Physical euclidean time coordinate " "along which correlation is measured."
    ),
    "Internal": (
        "Any further internal structure, e.g.,"
        "an index enumerating interpolating operators, "
        "a blocking or smearing level, "
        "or any combination thereof."
    ),
    "Correlation": "Measured values of the correlators.",
    "Vac_exp": "Measured values of the vacuum expectation values (VEVs).",
}
```

**Reply by edbennett** [reply]
Thanks—yes, I use `ruff format` via pre-commit and pre-commit.ci. The latter option would also have the benefit of my editor not fighting against it.
